packages/client/src/Client.ts

```diff
@@ -49,6 +56,16 @@ export interface Client extends SharedClient<MinimalClient> {
   <Q extends QueryDefinition<any, any, any>>(
     o: Q,
   ): QuerySignatureFromDef<Q>;
+
+  fetchMetadata<
+    Q extends Definition,
+  >(o: Q): Promise<
+    Q extends ObjectTypeDefinition<any, any> ? ObjectMetadata
+      : Q extends InterfaceDefinition<any, any> ? InterfaceMetadata
+      : Q extends ActionDefinition<any, any, any> ? ActionMetadata
+      : Q extends QueryDefinition<any, any, any> ? QueryMetadata
```

Comment:
I'm wondering after we get my thinned out objects and co merged in, if we should just be returning the `*Definition` for the metadata? So this woul dbe `Q extends MinObjectDef ? ObjectTypeDefinition<any,any> : ...`

Reply:
Yeah we can do that, my goal with this was to be able to easily control what we expose to the user. Based on what we discussed yesterday I dont want to return apiName, osdk metadata, or anything else thats on your MinObjectDef type because thats at compile. Thats mainly it, although if there's a case where we want something exposed on the type but not as a concrete value that adds to the argument of keeping it separate.

Reply:
Its probably easier for us to maintain the parity. The osdkmetadata is optional and the apIName they already know but it doesnt end the world to include it.